### ISSUE TITLE
makeRelative() symlink resolution

### ISSUE THREAD

**Issue by achubaty** [PredictiveEcology/reproducible#362]
downstream issue with `SpaDES.core::relativizePaths()` and resolving symlinks.

`reproducible:::makeRelative()` is used, which is the real culprit

_Originally posted by @achubaty in https://github.com/PredictiveEcology/SpaDES.core/issues/263#issuecomment-1743471579_

-----

When relativizing paths, resolving symlinks early results in the wrong behaviour when project uses symlink to e.g. point to a network drive:

```
$ cd ~/GitHub/myProject
$ ls -la ./outputs
lrwxrwxrwx 1 achubaty achubaty 32 May 20  2022 outputs -> /mnt/projects/myProject/outputs
```

All of the outputs in this case can be read using project-relative paths, and should not be made absolute, as this breaks transferability across machines.

**Comment by achubaty** [comment]
see https://github.com/PredictiveEcology/SpaDES.config/blob/development/R/helpers-paths.R#L27-L64 for an implementation that deals with this issue

**Comment by achubaty** [comment]
`makeRelative()`'s current behaviour, where it does a `normPath(absoluteBase)` but not `normPath(files)`, is heavily relied upon in `prepInputs()` and `Cache()`, that changing its behaviour breaks too many things. This is in part caused by a change in `normPath()` which now explicitly uses `getwd()` to expand `"."`.

I've implemented `normPathRel()` and `getRelative()` in 0af9327b3d286de49acadbfbf01066cd69f9200a which have the behaviour I'd expect, which I will use downstream in `SpaDES.core`.

From the updated `normPath()`/`normPathRel()` documentation:

```r
d> getwd()
[1] "/home/achubaty/Documents/GitHub/PredictiveEcology/reproducible"
d> normPathRel("potato/chips")
[1] "potato/chips"
d> normPath("potato/chips")
[1] "/home/achubaty/Documents/GitHub/PredictiveEcology/reproducible/potato/chips"
```

From the new `makeRelative()`/`getRelative()` example:

```r
if (!identical(.Platform$OS.type, "windows")) {
  ## create a project directory (e.g., on a hard drive)
  (tmp1 <- tempdir2("myProject", create = TRUE))

  ## create a cache directory elsewhere (e.g., on an SSD)
  (tmp2 <- tempdir2("my_cache", create = TRUE))

  ## symlink the project cache directory to tmp2
  ## files created here are actually stored in tmp2
  prjCache <- file.path(tmp1, "cache")
  file.symlink(tmp2, prjCache)

  ## create a dummy cache object file in the project cache dir
  (tmpf <- tempfile("cache_", prjCache))
  cat(rnorm(100), file = tmpf)
  file.exists(tmpf)
  normPath(tmpf) ## note the 'real' location (i.e., symlink resolved)

  getRelative(tmpf, prjCache) ## relative path
  getRelative(tmpf, tmp2) ## relative path

  makeRelative(tmpf, tmp2) ## abs path; tmpf and normPath(tmp2) don't share common path
  makeRelative(tmpf, prjCache) ## abs path; tmpf and normPath(tmp2) don't share common path
  makeRelative(normPath(tmpf), prjCache) ## rel path; share common path when both normPath-ed

  unlink(tmp1, recursive = TRUE)
  unlink(tmp2, recursive = TRUE)
}
```

**Comment by eliotmcintire** [comment]
@achubaty Is this fixed now?
